File: 0014/solution.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        prefix = strs[0]
        for i in range(1, len(strs)):
            new_prefix = []
            for j in range(len(strs[i])):
                if j >= len(prefix) or strs[i][j] != prefix[j]:
                    break
                else:
                    new_prefix.append(strs[i][j])
            prefix = "".join(new_prefix)
        return prefix

File: 0014/solution.go
package main

func longestCommonPrefix(strs []string) string {
	res := strs[0]
	for i, v := range strs[1:] {
		j := 0
		for i < len(v) && j < len(res) {
			if v[i] == res[j] {
				i++
				j++
			} else {
				break
			}
		}
		res = res[:j]
	}
	return res
}
